pkg/registry/apis/provisioning/jobs/migrate/worker.go

```diff
@@ -153,10 +165,24 @@ func (w *MigrationWorker) Process(ctx context.Context, repo repository.Repositor
 		}
 	}
 
-	// Now import from out local checkout
-	// TODO: can we skip this and write while exporting?
-	// YES... but :( the export with *history* does not know the current value
-	return worker.importFromRepo(ctx, w.storageStatus)
+	// Clear unified and allow writing
+	err = worker.wipeUnifiedAndSetMigratedFlag(ctx, w.storageStatus)
+	if err != nil {
+		return fmt.Errorf("unable to reset unified storage %w", err)
+	}
+
+	// enable sync (won't be saved)
+	repo.Config().Spec.Sync.Enabled = true
+
+	// Delegate the import to a sync (from the already checked out repository!)
+	sss := sync.NewSyncWorker(w.client, w.parsers, w.lister, w.storageStatus)
```

Comment:
let's not create this one here. Let's construct the migrate worker with this sync worker in `register.go`.